tpm2_pytss/utils.py

```diff
@@ -226,3 +227,117 @@ def unwrap(
         )
 
     return s
+
+
+def create_ek(ectx, ekalg=TPM2_ALG.ECC, session=ESYS_TR.NONE):
```

Comment:
I would change this slightly to get_ek_template and be able to pass session1, session2, and session3 here so folks pass everything into the esys calls and then I would have the createprimary call be by the user. This way the session1, which is auth for the ENDORSEMENT hierarchy can be set by the caller to an HMAC session, audit session, etc.

Reply:
Oh and add type hints to the parameters.

Reply:
docstring with link to the generic spec landing page:
  - https://trustedcomputinggroup.org/resource/tcg-ek-credential-profile-for-tpm-family-2-0/

Reply:
@idesai could you weigh in on this API design with your thoughts as well?

Reply:
I agree. Something to consider... because it uses a TCG EK credential specific value for authorization policy and attributes get_tcg_ek_template might be a good name.

Reply:
I kept only one session as the only usefulness I see of passing a session after removing create_primary is for audit and encrypt sessions.
As for the name, I prefer to keep it as close to tpm2_createek as reasonable, to keep it slightly familiar for tpm2-tools users, but I can go either way.

Reply:
You can have separate audit and encrypt sessions. I was tring to future proof the API a bit since session1,2,3 would just be the defaults and only if folks are interested in that control will they use them.

Reply:
OK, I'll change it to session1 and session2, but I'm worried about session3, the only call that would take all three is get_capability and that would break expectations for session3